### DISCUSSION TITLE
Is adding of approved directories to Butler config file for file system access with wildcard possible?

### DISCUSSION THREAD

**Discussion by rumen-vasilev** [ptarmiganlabs/butler#830] [Q&A] [answered]
Hello guys,

I want to use the REST API endpoint from Butler to delete QVD files in Windows directories. I have a lot of subdirectories and I don't want to insert every single subdirectory in the config file to approve it for the deletion. Is it possible to add a root directory with wildcard or something like that to save me the regular effort of maintaining the configuration file?

BR,
Rumen

**Comment by mountaindude** [comment, marked as answer]
Good question.

The theory is that you specify directories in the config file, then the file copy/move/delete is allowed in the specified directory and all subdirectories. 

But... When looking at the code just now I realised there is a bug in there.
The result is that the file operations will only work in the exact directory that is specified in the config file.

The bug is easy to fix so it will be included in next release.

**Reply by rumen-vasilev** [reply]
Thank you very much!

**Comment by mountaindude** [comment]
@rumen-vasilev Butler 9.2.2 was released a few minutes ago.
Available from the downloads page, https://github.com/ptarmiganlabs/butler/releases
It contains a fix for the bug involving subdirectories of approved directories for file operations.

Bottom line is that you should now only have to specify the root directory in which file copy/move/delete operations should be allowed, then those operations will also work in subdirectories. Give it a spin, I have tested it quite a bit and added automatic test cases for a bunch of scenarios, so it should work :)

**Reply by rumen-vasilev** [reply]
I tested the new version. It works! Thanks for the fast fixing!

**Comment by rumen-vasilev** [comment]
The issue was fixed in version 9.2.2
